services/app-web/src/pages/StateSubmission/RateDetails/V2/RateDetailsV2.tsx

```diff
@@ -286,104 +359,212 @@ const RateDetailsV2 = ({
-                                            }
-                                        )
-                                    } else {
-                                        navigate(
-                                            RoutesRecord.DASHBOARD_SUBMISSIONS
-                                        )
+                                    </FieldArray>
+                                </fieldset>
+                                <PageActions
+                                    pageVariant={
+                                        displayAsStandaloneRate
+                                            ? 'STANDALONE'
+                                            : undefined
                                     }
-                                }}
-                                saveAsDraftOnClick={
-                                    displayAsStandaloneRate
-                                        ? undefined
-                                        : async () => {
-                                              await handlePageAction(
-                                                  rateFormValues.rates,
-                                                  setSubmitting,
-                                                  {
-                                                      type: 'SAVE_AS_DRAFT',
-                                                      redirectPath:
-                                                          'DASHBOARD_SUBMISSIONS',
-                                                  }
-                                              )
-                                          }
-                                }
-                                disableContinue={
-                                    shouldValidate &&
-                                    !!Object.keys(errors).length
-                                }
-                                actionInProgress={isSubmitting}
-                            />
-                        </UswdsForm>
-                    </>
-                )
-            }}
-        </Formik>
+                                    backOnClick={async () => {
+                                        if (dirty) {
+                                            await handlePageAction(
+                                                rates,
+                                                setSubmitting,
+                                                {
+                                                    type: 'CANCEL',
+                                                    redirectPath:
+                                                        'DASHBOARD_SUBMISSIONS',
+                                                }
+                                            )
+                                        } else {
+                                            navigate(
+                                                RoutesRecord.DASHBOARD_SUBMISSIONS
```

Comment:
I noticed that when I tried to click continue from the Rates Details page I was routed to the dashboard instead of the Contacts page which is what I was expecting